artifactory/commands/transferfiles/transfer_test.go

```
@@ -0,0 +1,162 @@
+		if r.RequestURI == "/"+pluginsExecuteRestApi+"uploadChunk" {
+			w.WriteHeader(http.StatusAccepted)
+			content, err := json.Marshal(UploadChunkResponse{UuidTokenResponse: UuidTokenResponse{UuidToken: uuidTokenForTest}})
+			assert.NoError(t, err)
+			_, err = w.Write(content)
+			assert.NoError(t, err)
+		} else if r.RequestURI == "/"+pluginsExecuteRestApi+"getUploadChunksStatus" {
+			totalChunkStatusVisits++
+
+			// Read body
+			content, err := io.ReadAll(r.Body)
+			assert.NoError(t, err)
+			var actual UploadChunksStatusBody
+			assert.NoError(t, json.Unmarshal(content, &actual))
+
+			// Make sure all parameters as expected
+			assert.Len(t, actual.UuidTokens, 1)
+			assert.Equal(t, uuidTokenForTest, actual.UuidTokens[0])
+
+			resp := UploadChunksStatusResponse{ChunksStatus: []ChunkStatus{
+				{
+					UuidTokenResponse: UuidTokenResponse{UuidToken: uuidTokenForTest},
+					Status:            InProgress,
+				},
+			}}
+
+			// If already visited chunk status, return status done this time.
+			if totalChunkStatusVisits > 1 {
+				resp.ChunksStatus[0].Status = Done
+				resp.ChunksStatus[0].Files = []FileUploadStatusResponse{{FileRepresentation: fileSample, Status: Success, StatusCode: http.StatusOK}}
+			}
+			content, err = json.Marshal(resp)
+			assert.NoError(t, err)
+			_, err = w.Write(content)
+			assert.NoError(t, err)
+		}
+	})
+	defer testServer.Close()
+
+	srcPluginManager, err := createSrcRtUserPluginServiceManager(serverDetails)
+	if err != nil {
+		assert.NoError(t, err)
+		return
+	}
+
+	curThreads = 8
+	uploadTokensChan := make(chan string, 3)
+	doneChan := make(chan bool, 1)
+
+	// Sends chunk to upload, polls on chunk twice - once when it is still in progress, and once after it is done.
+	err = uploadChunkAndAddToken(srcPluginManager, UploadChunk{
+		UploadCandidates: []FileRepresentation{fileSample},
+	}, uploadTokensChan)
+	if err != nil {
+		assert.NoError(t, err)
+		return
```

```suggestion
```

---

It is on purpose so we will fail fast